portable/Zynq.2019.3/ff_sddisk.c

```diff
@@ -432,17 +432,19 @@ static CacheMemoryInfo_t * pucGetSDIOCacheMemory( BaseType_t xPartition )
 }
 /*-----------------------------------------------------------*/
 
-/* Initialise the SDIO driver and mount an SD card */
-BaseType_t xMountFailIgnore = 0;
+FF_Disk_t * FF_SDDiskInit( const char * pcName )
+{
+    return FF_SDDiskInitWithSettings( pcName, pdFALSE, 0U );
+}
 
-/* _HT_ : the function FF_SDDiskInit() used to mount partition-0.
- * It would be nice if it has a parameter indicating the partition
- * number.
- * As for now, the partion can be set with a global variable 'xDiskPartition'.
+/**
+ * Initialize the SD Disk with configurable settings
+ * @param[in] uxMountFailIgnore ignore fails on mount, set to true when have systems where may not have initialized mount
+ * @param[in] uxDiskPartition the disk partition number to use
  */
-BaseType_t xDiskPartition = 0;
-
-FF_Disk_t * FF_SDDiskInit( const char * pcName )
+FF_Disk_t * FF_SDDiskInitWithSettings( const char * pcName,
+                                       BaseType_t xMountFailIgnore,
+                                       BaseType_t xDiskPartition )
```

Comment:
One slight change to allow for future proofing this.

Suggest putting all of the settings into a struct:

```
typedef struct FFInitSettings_s
{
BaseType_t xMountFailIgnore;
BaseType_t xDiskPartition;
} FFInitSettings_t;
```

And then change the API to:

```c
FF_Disk_t * FF_SDDiskInitWithSettings( const char * pcName,
                                       const FFInitSettings_t * settings)
```

That way you can append to the settings later on without changing the API - as long as the defaults are something reasonable (most likely candidate) the API doesn't have to change for new initialization configuration settings.

@AniruddhaKanhere - you are also missing adding the API to the header file.

Reply:
@phelter , yes a good idea to put it into a struct. The only obligation will be to clear all (unused) fields in the struct. Thanks, Hein